src/internal/FrankenSubject.ts

```diff
@@ -0,0 +1,53 @@
+import { Observable } from './Observable';
+import { Observer, PartialObserver, ObservableInput } from './types';
+import { hostReportError } from './util/hostReportError';
+import { from } from './observable/from';
+
+/**
+ * Glues any observer to any observable to make a "subject".
+ *
+ * Assembles the parts like Frankenstein to make your own monster.
+ *
+ * Developer's note: `FrankenSubject` does NOT extend `Subject`, therefor
+ * `instanceof Subject` checks will not work.
+ */
+export class FrankenSubject<In, Out> extends Observable<Out> implements Observer<In> {
```

Comment:
Just thinking out loud, wouldn't it be a good idea to override the pipe method here to fix the types specially for subjects.

Today if you call `pipe` on a Subject it will still return a Subject even though the typing say it is an observable ...